packages/playground/src/explorer/farms.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <view-layout>
+    <v-row>
+      <v-col>
+        <node-filters v-model="filterInputs" v-model:valid="isValidForm" @update:model-value="inputFiltersReset" />
+        <FarmsTable :items="farms" :loading="loading" v-model:selectedFarm="selectedFarm" @open-dialog="openDialog" />
+      </v-col>
+    </v-row>
+    <farmDialog v-if="selectedFarm" :openDialog="isDialogOpened" :farm="selectedFarm" @close-dialog="closeDialog" />
+  </view-layout>
+</template>
+
+<script lang="ts" setup>
+import type { CertificationType, FarmsQuery, PublicIp } from "@threefold/gridproxy_client";
+import { onMounted, ref, watch } from "vue";
+import { useRoute } from "vue-router";
+
+import router from "@/router";
+import { inputsInitializer } from "@/utils/filter_farms";
+
+import { getFarmQueries, getFarms } from "./utils/helpers";
+import {
+  type FarmFilterInputs,
+  type FarmFilterOptions,
+  type FarmMixedFilter,
+  farmOptionsInitializer,
+} from "./utils/types";
+
+export interface IFarm {
+  farmId: number;
+  name: string;
+  twinId: number;
+  pricingPolicyId: number;
+  certificationType: CertificationType;
+  publicIps: PublicIp[];
+  stellarAddress?: string;
+  totalPublicIp: number;
+  usedPublicIp: number;
+  freePublicIp: number;
+}
+const route = useRoute();
+
+const loading = ref<boolean>(false);
+const farms = ref<IFarm[]>([]);
+const isDialogOpened = ref<boolean>(false);
+const selectedFarm = ref<IFarm>();
+const filterInputs = ref<FarmFilterInputs>(inputsInitializer);
+const filterOptions = ref<FarmFilterOptions>(farmOptionsInitializer);
+const isValidForm = ref<boolean>(false);
+const farmsCount = ref<number>(0);
+const mixedFilters = ref<FarmMixedFilter>({
+  inputs: filterInputs.value,
+  options: filterOptions.value,
+});
+
+const _getFarms = async (queries: Partial<FarmsQuery>) => {
+  loading.value = true;
+  try {
+    const { count, data } = await getFarms(queries);
+    if (count) {
+      farmsCount.value = count;
+    }
+    farms.value = data.map(farm => {
+      const ips = farm.publicIps;
+      const total = ips.length;
+      const used = ips.filter(x => x.contract_id === 0).length;
+      return {
+        ...farm,
+        totalPublicIp: total,
+        usedPublicIp: used,
+        freePublicIp: total - used,
+      };
+    });
+  } catch (error) {
+    console.log(error);
+  } finally {
+    loading.value = false;
+  }
+};
+
+const checkPath = async () => {
+  if (route.query.twinId) {
+    router.replace(route.path);
+  }
+};
+
+const openDialog = (item: IFarm) => {
+  selectedFarm.value = item;
+  isDialogOpened.value = true;
+};
+
+const closeDialog = () => {
+  if (route.query.twinId) {
+    router.replace(route.path);
+  }
+  isDialogOpened.value = false;
+};
+
+watch(
+  mixedFilters,
+  async () => {
+    const queries = getFarmQueries(mixedFilters.value);
+    await _getFarms(queries);
+  },
+  { deep: true },
+);
```

Comment:
I think it needs more efforts

1. we need to change the `mixedFilter` since we'll delete it and use the `inputFilters` instead.
2. we have to wait on request for 1 or 2 seconds when the user stops writing, please see the attached record.
3. we need to disable the form fields while requesting the gridProxy, please if you implemented the filter forms from scratch you don't need to do it, you can use the implemented one.
4. display a message to the user to inform them that the request will happen after 1 second. see the `nodes` view.
5. we must make the request only if there are no errors. 
6. the pagination seems to be doesn't work as expected | `Items per page:`, it even doesn't reflect the selection.

[Screencast from 10-22-2023 06:29:06 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/8b501003-9609-4d2a-b487-6a096d4d31e4)

instead of it you can use the `debounce` method to make your request, see this example

```vue
<script lang="ts">
  const request = debounce(_requestNodes, 1000);
  watch(
      inputFilters,
      async () => {
        await request(attrs);
      },
      { deep: true },
    );
</script>
```
The idea is we delay the request for one second after the user stops writing to avoid making many requests to the grid proxy.